### DISCUSSION TITLE
Is there a way to implement an internal SRFF ?

### DISCUSSION THREAD

**Discussion by arnonsha** [SystemRDL/systemrdl-compiler#194] [Q&A] [answered]
Hi
One of my fields needs to be implemented using SRFF: read/written by SW, reset asynchronously by the bus reset and set asynchronously by HW.
Is it possible to do this internally using RDL, or the only way to have such filed is by implementing it externally ?

thanks

**Comment by amykyta3** [comment, marked as answer]
RDL lets you define an asynchronous reset signal + value, but that will only let you perform an asynchronous set **or** clear. There is also the hwset/hwclr property, but those are implied to be synchronous.
I don't think there is a way to get both asynchronous set & clear at the same time for a field given the native RDL properties defined by the spec.

Agree that implementing it externally is the way to go.

**Comment by arnonsha** [comment]
Thank you for confirming.
I'll implement using an external register.
